### DISCUSSION TITLE
Windows ssl connection reset

### DISCUSSION THREAD

**Discussion by ghost** [apache/couchdb#3267] [Q&A] [answered]
## Description
So I am running couchdb on windows10 pro, installed via msi v3.1.1, after I generated selfsigned certs, updated local.ini and restarted service requests to http://127.0.0.1:6984 end up with connection reset error. I tried to generate same certificates on linux with same version of couchdb and it worked right away.

I also tried to find any relevant logs for couchdb working, but only one file looks like log is C:\Program Files\Couch DB\var\logs\couchdb, which always have only one line: New log. Changing log level not helped a bit, looking into AppData direcrtories for all users not showed any sign of couchdb generated folders with data (must be I missed something here as db have to save its files somewhere!)

## Steps to Reproduce

msiexec /i apache-couchdb-3.1.1.msi /quiet ADMINUSER=admin ADMINPASSWORD=hunter2 /norestart
Generate certs and CA according to following guide
https://gist.github.com/fntlnz/cf14feb5a46b2eda428e000157447309#create-a-certificate-done-for-each-server
uncomment [ssl] section in local.ini, set enable true and fill it with cacert_file, cert_file and key_file values
Restart couchdb
curl http://127.0.0.1:5984
curl https://127.0.0.1:6984

## Expected Behaviour

On windows 10 ssl should work same normally as on linux and  logs produced telling if ssl files found, loaded or there some problemd with them.

## Your Environment

* CouchDB version used: 3.1.1
* Browser name and version: chrome latest (for request used curl from git bash and powershel Invoke-RestMethod)
* Operating system and version: Windows 10 Pro
OpenSSL v1.1.1i-dev

## Additional Context

All actions with couchdb made with administrator rights (couchdb install, ssl certs genertion)

**Comment by wohali** [comment]
Hi @Cherviakov ,

The issue is that CouchDB (and Erlang) expects keys in a very specific format.

If you try the walkthrough [in our documentation](https://docs.couchdb.org/en/stable/config/http.html?highlight=ssl#https-ssl-tls-options), does it still fail for you?

**Comment by ghost** [comment]
Tired exactly like in docs, got same problem.

**Comment by ghost** [comment]
Also https://www.openssl.org/docs/HOWTO/certificates.txt this link give error 404 page not found.

**Comment by ghost** [comment, marked as answer]
Made some more experiments, it appears to be problem with path in Windows containing spaces, like "C:\Program Files", when moved to "C:\Users\Me" then all worked.

But then how to sepcify in windows path with spaces as there it is quite a usual matter? Please provide examples for windows as well.

Also still not sure where to check couchdb logs in windows

And finally, format of ssl certs requires openssl to be installed, please specify version of openssl to use (would be good with link to download for windows). If there actually way to use New-SelfSignedCertificate cmdlet it would be even better, as it is present out of the box on windows.
